roles/module_openapi_cloud/templates/module_directory/cisco_intersight/default_module.j2

```diff
@@ -0,0 +1,331 @@
+        except KeyError:
+            _json = {}
+        return await update_changed_flag(_json, resp.status, "delete")
+
+
+
+{% elif operation in ["create", "clone", "instant_clone"] -%}
+async def _{{ operation }}(params, session):
+
+{% if name.startswith("content_") %}
+    {% set haz_lookup = 1 %}
+    lookup_url = per_id_url = build_url(params)
+    uniquity_keys = ["name"]
+    comp_func = None
+{% elif name == "vcenter_vmtemplate_libraryitems" %}
+    {% set haz_lookup = 1 %}
+    lookup_url = "https://{api_uri}/api/content/library/item?library_id={library}".format(**params)
+    per_id_url = "https://{api_uri}/api/content/library/item".format(**params)
+    uniquity_keys = ["name"]
+    comp_func = None
+{% elif name == "vcenter_ovf_libraryitem" %}
+    {% set haz_lookup = 1 %}
+    library_id = params['target']['library_id'] if 'library_id' in params["target"] else None
+    lookup_url = f"https://{params['api_uri']}/api/content/library/item?library_id={library_id}"
+    per_id_url = "https://{api_uri}/api/content/library/item".format(**params)
+    uniquity_keys = None
+    def comp_func(device):
+        return device["value"]["name"] == params["create_spec"].get("name")
+{% elif list_index %}
+    {% set haz_lookup = 1 %}
+    {% set haz_lookup_with_filters = 1 %}
+    lookup_url = per_id_url = build_url(params)
+    uniquity_keys = ["{{list_index}}"]
+    comp_func = None
+
+    async def lookup_with_filters(params, session, url):
+        search_filter = ''
+        {% if name in ["vcenter_host", "vcenter_datacenter"] %}
+        if "folder" not in params:
+            return
+        search_filter = f"&folders={params.get('folder')}"
+        {% endif %}
+        if "name" not in params:
+            return
+        async with session.get(f"{url}?names={params['name']}{search_filter}") as resp:
+            _json = await resp.json()
+            if isinstance(_json, list) and len(_json) == 1:
+                return await get_device_info(session, url, _json[0]["{{list_index}}"])
+{% else %}
+    uniquity_keys = []
+{% endif %}
+
+{% if haz_lookup is defined %}
+    _json = None
+    {% if list_index %}
+    if params["{{list_index}}"]:
+        _json = await get_device_info(session, build_url(params), params["{{list_index}}"])
+    {% endif %}
+    if not _json and (uniquity_keys or comp_func):
+        _json = await exists(params, session, url=lookup_url, uniquity_keys=uniquity_keys, per_id_url=per_id_url, comp_func=comp_func)
+    {% if haz_lookup_with_filters is defined %}
+    if not _json:
+        _json = await lookup_with_filters(params, session, build_url(params))
+    {% endif %}
+
+    if _json:
+        if "value" not in _json:  # 7.0.2+
+            _json = {"value": _json}
+        if "_update" in globals():
+            params["{{list_index}}"] = _json["id"]
+            return (await globals()["_update"](params, session))
+
+
+	{% if name == "vcenter_vmtemplate_libraryitems" %}
```

Comment:
This block is vmware specific. This has to be modified to support to cisco.intersight.